lib/db.rb

```diff
@@ -1,27 +1,30 @@
 class DB
 
-  def initialize(contact_list = [])
-    @contact_list = contact_list
+  def initialize(reader = nil, writer = nil)
+    @reader = reader
+    @writer = writer
   end
 
   def size
-    contact_list.size
+    all.size
   end
 
   def add(contact)
-    contact_list << contact
+    contacts = all
+    contacts << contact
+    writer.write_contacts(contacts) if !writer.nil?
```

Comment:
lol
"- *usted no nada nada?*"
"- *no, no traje traje*"

I'll try to think of a better name :+1: